### PR TITLE
Added soft delete API endpoint for file uploads

### PR DESCRIPTION
**Pull request by Marmik2003**
Added soft delete API endpoint for issue https://github.com/coronasafe/care_fe/issues/2288

### CODE REVIEW

**Comment by vigneshhari** [comment]
Only keep changes relevant to the PR, either remove the other changes or rebase with master if the changes are merged in

File: care/facility/api/viewsets/file_upload.py [hunk 1]
```diff
@@ -33,6 +34,8 @@ def get_serializer_class(self):
             return FileUploadListSerializer
         elif self.action == "retrieve":
             return FileUploadRetrieveSerializer
+        elif self.action == "destroy":
```
Comment (vigneshhari) [inline care/facility/api/viewsets/file_upload.py:37]: Why create a separate serializer?

File: care/facility/api/serializers/file_upload.py [hunk 2]
```diff
@@ -117,3 +117,9 @@ class Meta:
         model = FileUpload
         fields = ("id", "name", "uploaded_by", "created_date", "read_signed_url", "file_category")
         read_only_fields = ("associating_id", "name", "created_date")
+
+
+class FileUploadDestroySerializer(serializers.ModelSerializer):
```
Comment (vigneshhari) [inline care/facility/api/serializers/file_upload.py:122]: How exactly is this serializer supporting soft deletions?

**Review by vigneshhari — CHANGES_REQUESTED** [review]

**Review by sainAk — APPROVED** [review]

**Comment by sainAk** [comment]
Ignore the review it was for a test

**Review by sainAk — APPROVED** [review]
This is a test review

**Comment by Marmik2003** [comment]
closing PR for removing unnecessary commits

**Pull request closed without merging**